### DISCUSSION TITLE
How to use ghidra to debug on switch ?

### DISCUSSION THREAD

**Discussion by PoloNX** [NationalSecurityAgency/ghidra#5417] [Debugger]
Hi, I'm here to ask you how to debug my switch homebrew on ghidra.

In fact I have a crash at one moment and I want to debug my app in aim to catch the issue. I have a gdb server on my switch with atmosphere and I already connected my switch to gdb in cli mod (https://gist.github.com/jam1garner/c9ba6c0cff150f1a2480d0c18ff05e33) But it's not very easy to understand so I wanted to use one app like ghidra to debug my app.

Devkitpro provide one gdb build for nintendo switch called aarch64-none-elf-gdb.exe

![javaw_1cs2Dh5cp7](https://github.com/NationalSecurityAgency/ghidra/assets/57038157/c5306502-61d2-4ab8-8c08-8a0fefd77dcc)
![javaw_PO2Ia293en](https://github.com/NationalSecurityAgency/ghidra/assets/57038157/23efeb78-001f-4c21-ba48-1b5457e8489e)

Now I'm here and I have this error. Someone can help me ?

**Comment by d-millar** [comment]
Try the following:

(1) From the Debugger Tool's "Debugger Targets" window,  choose "gdb" (or GNU gdb local IN-VM on older versions) and for "GDB launch command" enter "aarch64-none-elf-gdb.exe".
(2) The interpreter window should start up.  From its command line, enter "target remote 192.168.137.229:22225".
(3) Depending on your gdbserver, you may need to do some additional work to get a Module List (which is necessary for syncing the Dynamic and Static Listings).

Let us know if you get that far and will send more info (or check out the many related posts in Issues/Discussions).
